src/sentry/tasks/post_process.py

```diff
@@ -1472,6 +1472,26 @@ def detect_new_escalation(job: PostProcessJob):
         return
 
 
+GROUP_VALIDATION_CACHE_KEY = lambda project_id: f"detect-invalid-group-status-{project_id}"
+GROUP_VALIDATION_CACHE_DURATION = 60 * 60 * 24 * 3  # 3 days
+
+
+def detect_invalid_group_status(job: PostProcessJob):
+    from sentry.tasks.detect_invalid_group_status import detect_invalid_group_status
+
+    project_id = job["event"].project.id
+    cache_key = GROUP_VALIDATION_CACHE_KEY(project_id)
+
+    if cache.get(cache_key):
+        return
+
+    try:
+        cache.set(cache_key, True, GROUP_VALIDATION_CACHE_DURATION)
+        detect_invalid_group_status.delay(project_id=project_id)
```

Comment:
@snigdhas do we need to check the project every time we process an event? That seems quite expensive. Maybe there is a way we can be smarter about running the validation?

Reply:
This would check the project for the first event every 3 days because of the GROUP_VALIDATION_CACHE_DURATION, but I agree that there's probably a smarter way to go about this. Running a regular task, like the auto_update, seemed more expensive and likely to fail in the same way